app/models/concept.rb

```diff
@@ -3,14 +3,23 @@ class Concept < ActiveRecord::Base
     validates :name, presence: true, uniqueness: true
     validates :description, presence: true
     has_many :messages
-    has_many :tag2concepts
+    has_many :tag2concepts, :dependent => :destroy
     has_many :tags, :through => :tag2concepts
     include ActiveModel::Serialization
     
     def attributes
         {:name => name, :description => description}    
     end
     
+    def self.verify_row(name, description)
```

Comment:
We should verify not based on nil, but verify based on if it has a nonspace character